react/Components/toastArea.tsx

```diff
@@ -0,0 +1,113 @@
+import { toast } from '../utils/definedMessages'
+
+const ToastArea = () => {
+  const DEFAULT_ALERT_AUTOCLOSE_TIMEOUT_MS = 3000
+
+  const intl = useIntl()
+
+  const { showAlert, setShowAlert, rescue, code } = useStore()
+
+  function changeValueShowAlert() {
+    if (showAlert !== ShowAlertOptions.notShow) {
+      setShowAlert(ShowAlertOptions.notShow)
+    }
+  }
+
+  useEffect(() => {
+    const interval = setInterval(
+      changeValueShowAlert,
+      DEFAULT_ALERT_AUTOCLOSE_TIMEOUT_MS
+    )
+
+    return () => {
+      clearInterval(interval)
+    }
+  }, [showAlert])
+
+  if (showAlert === ShowAlertOptions.alertSave) {
+    return (
+      <div className={`notification-container bottom-left`}>
+        <div className={`notification bg-color-sucess toast bottom-left`}>
+          <div className="notification-image">
+            <img src={success} alt="" />
+          </div>
+          <div>
+            <p className="notification-title">
+              {intl.formatMessage(toast.rescueValue) +
+                rescue.toLocaleString('pt-br', { minimumFractionDigits: 2 })}
+            </p>
+            <p className="notification-message">
+              {intl.formatMessage(toast.rescueSucess)}
+            </p>
+          </div>
+        </div>
+      </div>
+    )
+  }
+
+  if (showAlert === ShowAlertOptions.alertCopySuccess) {
+    return (
+      <div className={`notification-container bottom-left`}>
+        <div className={`notification bg-color-sucess toast bottom-left`}>
+          <div className="notification-image">
+            <img src={success} alt="" />
+          </div>
+          <div>
+            <p className="notification-title">
+              {intl.formatMessage(toast.code) + code}
+            </p>
+            <p className="notification-message">
+              {intl.formatMessage(toast.copy)}
+            </p>
+          </div>
+        </div>
+      </div>
+    )
+  }
+
+  if (showAlert === ShowAlertOptions.alertError) {
+    return (
+      <div className={`notification-container bottom-left`}>
+        <div className={`notification bg-color-error toast bottom-left`}>
+          <div className="notification-image">
+            <img src={error} alt="" />
+          </div>
+          <div>
+            <p className="notification-middle">
+              {intl.formatMessage(toast.rescueError)}
+            </p>
+          </div>
+        </div>
+      </div>
+    )
+  }
+
+  if (showAlert === ShowAlertOptions.alertCopyError) {
+    return (
+      <div className={`notification-container bottom-left`}>
+        <div className={`notification bg-color-error toast bottom-left`}>
+          <div className="notification-image">
+            <img src={error} alt="" />
+          </div>
+          <div>
+            <p className="notification-middle">
+              {intl.formatMessage(toast.copyError)}
+            </p>
+          </div>
+        </div>
+      </div>
+    )
+  }
+
+  return null
```

Comment:
I think that you could change this component. 
You have some parts that reperts itself so you could change to just write that once some thing like:

```
if (showAlert !== ShowAlertOptions.noShow) {
    return (
      <div className={`notification-container bottom-left`}>
        <div className={functionToSetClasses()}>
          <div className="notification-image">
            <img src={functionToShowIfErrorImageOrSuccess()} alt="" />
          </div>
          {funtionThatRendersWhatIsCustomToEachErrorWithYourIfs()}
        </div>
      </div>
    )
 }
  
  return null
```

**funtionThatRendersWhatIsCustomToEachErrorWithYourIfs**: This function would have a logic similar to your ifs right now, but it would only return the custom part to each type of alert.

**functionToShowIfErrorImageOrSuccess**: This function would have a similar logic to show the success or error image depending on the showAlert option

**functionToSetClasses**: This function would return `notification bg-color-error toast bottom-left` or `notification bg-color-sucess toast bottom-left` based on the showAlert option

Reply:
Yeah! I reaaly agri with you @gabrielHosino 
Another way to improve this is using `dictionary`. Where the conditional `if` is the key of enum, and the `classes`, `texts` and `image` is the valeu of this key. 

```
interface AlertInfo {
    className: string
    imageSrc: string
    message: string
}

const dictionary: { [key: number]: User } = {
   0: { className: 'myClass', imageSrc: 'success', message: 'success message'},
   1: { className: 'myClass', imageSrc: 'error', message: 'errormessage'},
};

const alertInfo = dictionary[ShowAlertOptions.alertCopyError]

// console.log(alertInfo.className)
// console.log(alertInfo.imageSrc)
// console.log(alertInfo.message)
```

You can see more about dictionary approach here: https://blog.logrocket.com/building-type-safe-dictionary-typescript/

Reply:
Love this suggestion ❤️

Reply:
@wagnerlduarte I think I understand
I will try to do this!
Thank by this suggestion :)